### PR TITLE
 Add status subresource and conditions to EtcdStorage CRD

### LINKED ISSUE #8: Utilize the subresource status capability of the CRDs when updating CRD status
See [#2/comment](https://github.com/xmudrii/etcdproxy-controller/pull/2/files#r192097283) for more details.

### PR DESCRIPTION
**Pull request by xmudrii**
This PR adds the Status subresource with the condition describing the current status of an `EtcdStorage` instance (currently shows only is it deployed).

Closes #8

### CODE REVIEW

File: pkg/controller/etcdproxy/etcdproxy_controller.go [hunk 1]
```diff
@@ -307,6 +307,19 @@ func (c *EtcdProxyController) syncHandler(key string) error {
 		return err
 	}
 
+	if !etcdstoragev1alpha1.IsEtcdStorageConditionTrue(etcdstorage, etcdstoragev1alpha1.Deployed) {
```
Comment (deads2k) [inline pkg/controller/etcdproxy/etcdproxy_controller.go]: This seems weird.  Break it out into a separate pull for discussion please.

**Comment by deads2k** [comment]
API looks ok (we can talk about specific conditions later), but please split out the bit I commented on.

**Comment by xmudrii** [comment]
@deads2k Sounds good to me. Addressed the comment.

**Comment by deads2k** [comment]
lgtm

**Pull request merged**